lib/matplotlib/testing/compare.py

```diff
@@ -128,6 +133,109 @@ def convert(old, new):
+                return False
+            if self._proc.poll() is not None:
+                # Inkscape exited.
+                return False
+            if window == prompt:
+                # Successfully read until prompt.
+                return True
+
+    def __call__(self, orig, dest):
+        if (not self._proc  # First run.
+                or self._proc.poll() is not None):  # Inkscape terminated.
+            env = os.environ.copy()
+            # If one passes e.g. a png file to Inkscape, it will try to
+            # query the user for conversion options via a GUI (even with
+            # `--without-gui`).  Unsetting `DISPLAY` prevents this (and causes
+            # GTK to crash and Inkscape to terminate, but that'll just be
+            # reported as a regular exception below).
+            env.pop("DISPLAY", None)  # May already be unset.
+            # Do not load any user options.
+            # `os.environ` needs native strings on Py2+Windows.
+            env[str("INKSCAPE_PROFILE_DIR")] = os.devnull
+            # Old versions of Inkscape (0.48.3.1, used on Travis as of now)
+            # seem to sometimes deadlock when stderr is redirected to a pipe,
+            # so we redirect it to a temporary file instead.  This is not
+            # necessary anymore as of Inkscape 0.92.1.
+            self._stderr = TemporaryFile()
+            self._proc = subprocess.Popen(
+                [str("inkscape"), "--without-gui", "--shell"],
+                stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                stderr=self._stderr, env=env)
+            if not self._read_to_prompt():
+                raise OSError("Failed to start Inkscape")
+
+        # Inkscape uses glib's `g_shell_parse_argv`, which has a consistent
+        # behavior across platforms, so we can just use `shlex.quote`.
+        try:
+            fsencode = os.fsencode
+        except AttributeError:  # Py2.
+            def fsencode(s):
+                return s.encode(sys.getfilesystemencoding())
+
+        orig_b, dest_b = map(_shlex_quote_bytes, map(fsencode, [orig, dest]))
+        if b"\n" in orig_b or b"\n" in dest_b:
+            # Who knows whether the current folder name has a newline, or if
+            # our encoding is even ASCII compatible...  Just fall back on the
+            # slow solution.
+            return make_external_conversion_command(lambda old, new: [
+                str('inkscape'), '-z', old, '--export-png', new])(orig, dest)
+        self._proc.stdin.write(orig_b + b" --export-png=" + dest_b + b"\n")
+        self._proc.stdin.flush()
+        if not self._read_to_prompt():
+            # Inkscape's output is not localized but gtk's is, so the
+            # output stream probably has a mixed encoding.  Using
+            # `getfilesystemencoding` should at least get the filenames
+            # right...
+            self._stderr.seek(0)
+            raise ImageComparisonFailure(
+                self._stderr.read().decode(
```

Comment:
```
In [1]: from matplotlib.testing.compare import _SVGConverter; _SVGConverter()("/tmp/quux.svg", "/tmp/foo.png")
---------------------------------------------------------------------------
ImageComparisonFailure                    Traceback (most recent call last)
<ipython-input-2-c896ab68b2ce> in <module>()
----> 1 from matplotlib.testing.compare import _SVGConverter; _SVGConverter()("/tmp/quux.svg", "/tmp/foo.png")

/home/antony/src/extern/matplotlib/lib/matplotlib/testing/compare.py in __call__(self, orig, dest)
    225             raise ImageComparisonFailure(
    226                 self._stderr.read().decode(
--> 227                     sys.getfilesystemencoding(), "replace"))
    228 
    229     def __del__(self):

ImageComparisonFailure: ** Message: /dev/null is not a valid directory.
** Message: Inkscape will run with default settings, and new settings will not be saved. 

** (inkscape:25211): WARNING **: Could not create extension error log file '/dev/null/extension-errors.log'

** (inkscape:25211): WARNING **: Can't open file: /tmp/quux.svg (doesn't exist)

** (inkscape:25211): WARNING **: Can't open file: /tmp/quux.svg (doesn't exist)

** (inkscape:25211): WARNING **: Specified document /tmp/quux.svg cannot be opened (does not exist or not a valid SVG file)
```

The first few warnings should be spurious (due to intentionally setting `INKSCAPE_PROFILE_DIR` to a nonexistent folder) and I could technically crop them out, but I'd rather have the full info available just in case...